wwwroot/assets/scripts/utils.js

```diff
@@ -0,0 +1,73 @@
+const $ = document.querySelector.bind(document);
+import * as g from './global';
+
+export const getSVG = async (url, type) => {
+  try {
+    let svg = await ajax(url, type);
+    return svg;
+  }
+  catch(err) {
+    log('error while retrieving svg: ' + err.message);
+  }
+};
+
+const ajax = (url, type) => {
+  return new Promise(function(resolve, reject) {
+    let xhr = new XMLHttpRequest();
+    xhr.onload = function() {
+      if (type == 'text') resolve(this.responseText);
+      else if (type == 'XML') resolve(this.responseXML);
+    };
+    xhr.onerror = reject;
+    xhr.open('GET', url);
+    xhr.send();
+  });
+};
+
+export const displaySpinner = (message) => {
+  $('#blank').style.display = 'block';
+  $('#spinner-text').innerHTML = message;
+  $('#spinner-wrapper').style.display = 'block';
+};
+
+export const hideSpinner = () => {
+  $('#spinner-text').innerHTML = '';
+  $('#spinner-wrapper').style.display = 'none';
+  $('#blank').style.display = 'none';
+};
+
+export const displayError = (message) => {
+  displaySpinner(message);
+  setTimeout( () => {
+    hideSpinner();
+  },2000);
+};
+
+export const formatNumber = (n) => {
+  if (typeof n == 'number') {
```

Comment:
When you have more than 3 if/else is a smell that you need to split or rethink how to do it better.